sky/clouds/service_catalog/gcp_catalog.py

```diff
@@ -433,6 +448,30 @@ def list_accelerators(
                     price=info.price + vm_price,
                     spot_price=info.spot_price + vm_spot_price,
                 ))
+    # Sort the instances again after adding the vm price, as the order may
+    # have changed.
+    for acc_name, acc_infos in new_infos.items():
+        acc_infos.sort(
+            key=lambda info: (info.accelerator_count, info.instance_type, (
+                info.cpu_count if not pd.isna(info.cpu_count) else 0), info.
+                              price, info.spot_price, info.region))
+        if not all_regions:
+            # Only keep the cheapest instance type across all regions.
+            new_acc_infos = []
+            cur_key = None
+
+            def _get_key(info):
+                return (info.accelerator_name, info.accelerator_count,
+                        info.instance_type,
+                        (info.cpu_count if not pd.isna(info.cpu_count) else 0))
```

Comment:
```suggestion
                return (info.accelerator_count,
                        info.instance_type,
                        (info.cpu_count if not pd.isna(info.cpu_count) else 0))
```
nit: do we really need the ACC name? Reason to do the refactoring: I was confused when I first time saw the key (it seems different w/ previous sort key but essentially the same IIUC?)

Reply:
After the refactoring it will be the first, second and third sort keys used in the `sort` method and if all of them are the same, it is sorted as the fourth key, i.e. `info.price`

Reply:
Ahh, good catch! We should leave out the `acc_name` here. Just updated. Thanks!